devmand/gateway/src/devmand/test/cli/CliScaleTest.cpp

```diff
@@ -77,47 +65,52 @@ const static int REQUESTS = 10;
-    const std::shared_ptr<SshSessionAsync>& session =
-        std::make_shared<SshSessionAsync>(to_string(port), executor);
-
-    session->openShell("172.8.0.100", port, "cisco", "cisco").get();
-
-    shared_ptr<CliFlavour> cl = CliFlavour::create("ubiquiti");
-
-    const shared_ptr<PromptAwareCli>& cli =
-        std::make_shared<PromptAwareCli>(session, cl);
-
-    cli->initializeCli();
-    cli->resolvePrompt();
-    event* sessionEvent = SshSocketReader::getInstance().addSshReader(
-        readCallback, session->getSshFd(), session.get());
-    MLOG(MWARNING) << "Connecting device at port" << port
-                   << " at FD: " << session->getSshFd();
-    session->setEvent(sessionEvent);
-    return std::make_shared<QueuedCli>(to_string(port), cli, executor);
+  function<shared_ptr<Cli>(int)> connect = [this](int port) {
+    IoConfigurationBuilder ioConfigurationBuilder(
+        IoConfigurationBuilder::makeConnectionParameters(
+            to_string(port),
+            "172.8.0.1",
+            "cisco",
+            "cisco",
+            "ubiquiti",
+            port,
+            10s,
+            60s,
+            10s,
+            30,
+            cliEngine->getTimekeeper(),
+            cliEngine->getExecutor(Engine::executorRequestType::sshCli),
+            cliEngine->getExecutor(Engine::executorRequestType::paCli),
+            cliEngine->getExecutor(Engine::executorRequestType::rcCli),
+            cliEngine->getExecutor(Engine::executorRequestType::ttCli),
+            cliEngine->getExecutor(Engine::executorRequestType::qCli),
+            cliEngine->getExecutor(Engine::executorRequestType::rCli),
+            cliEngine->getExecutor(Engine::executorRequestType::kaCli)));
+    return ioConfigurationBuilder.createAll(ReadCachingCli::createCache());
   };
-  const Command& cmd = Command::makeReadCommand("show running-config");
+  const ReadCommand& cmd = ReadCommand::create("show running-config");
 
   {
-    std::chrono::steady_clock::time_point begin =
-        std::chrono::steady_clock::now();
-    vector<Future<shared_ptr<QueuedCli>>> connects;
+    chrono::steady_clock::time_point begin = chrono::steady_clock::now();
+    vector<Future<shared_ptr<Cli>>> connects;
     for (int cPort = START_PORT; cPort < DEVICES + START_PORT;
          cPort = cPort + 1) {
       MLOG(MWARNING) << "Connecting device at port: " << cPort;
-      Future<shared_ptr<QueuedCli>> future =
-          via(testExecutor.get(),
-              [&connects, connect, cPort]() { return connect(cPort); });
-      connects.push_back(std::move(future));
+      connects.emplace_back(connect(cPort));
     }
 
-    vector<shared_ptr<QueuedCli>> clis;
+    vector<shared_ptr<Cli>> clis;
     for (uint i = 0; i < connects.size(); i++) {
-      clis.push_back(std::move(connects.at(i)).get());
+      clis.push_back(move(connects.at(i)).get());
+
+      // Wait till connected. Is there a better way ?
+      while (clis.at(clis.size() - 1)
```

Comment:
I have an EXPECT_BECOMES_TRUE macro what might be a good use here. It will wait but only for a max time bound. We could add another time bound if that current one is too low.